pkg/utils/gardener/garden.go

```diff
@@ -363,3 +368,86 @@ func IsServedByGardenerAPIServer(resource string) bool {
 func IsServedByKubeAPIServer(resource string) bool {
 	return !IsServedByGardenerAPIServer(resource)
 }
+
+// ComputeRequiredExtensionsForGarden computes the extension kind/type combinations that are required for the
+// garden reconciliation flow.
+func ComputeRequiredExtensionsForGarden(garden *operatorv1alpha1.Garden, extensions []operatorv1alpha1.Extension) sets.Set[string] {
+	requiredExtensions := sets.New[string]()
+
+	if helper.GetETCDMainBackup(garden) != nil {
+		requiredExtensions.Insert(ExtensionsID(extensionsv1alpha1.BackupBucketResource, garden.Spec.VirtualCluster.ETCD.Main.Backup.Provider))
+	}
+
+	for _, provider := range helper.GetDNSProviders(garden) {
+		requiredExtensions.Insert(ExtensionsID(extensionsv1alpha1.DNSRecordResource, provider.Type))
+	}
+
+	for _, extension := range garden.Spec.Extensions {
+		requiredExtensions.Insert(ExtensionsID(extensionsv1alpha1.ExtensionResource, extension.Type))
+	}
+
+	return requiredExtensions
+}
+
+// CheckRuntimeExtensionInstallation checks if an Extension has been marked as "successfully" in the Garden runtime cluster.
+func CheckRuntimeExtensionInstallation(ctx context.Context, c client.Client, gardenNamespace, extensionName string) error {
```

Comment:
"Check" is not really self-explaining

```suggestion
func IsRuntimeExtensionInstallationSuccessful(ctx context.Context, c client.Client, gardenNamespace, extensionName string) error {
```

WDYT?